src/ResourceManager/GuestConfiguration/Commands.GuestConfiguration/Cmdlets/GetAzureRmVMGuestPolicyReport.cs

```diff
@@ -0,0 +1,180 @@
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.VmNameScope, Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = ParameterHelpMessages.VMName)]
+        [Parameter(ParameterSetName = ParameterSetNames.InitiativeIdScope, Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = ParameterHelpMessages.VMName)]
+        [Parameter(ParameterSetName = ParameterSetNames.InitiativeNameScope, Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = ParameterHelpMessages.VMName)]
+        [ValidateNotNullOrEmpty]
+        public string VMName { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.InitiativeNameScope, Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = ParameterHelpMessages.InitiativeName)]
+        [ValidateNotNullOrEmpty]
+        public string InitiativeName { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.InitiativeIdScope, Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = ParameterHelpMessages.InitiativeId)]
+        [ValidateNotNullOrEmpty]
+        public string InitiativeId { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.ReportIdScope, Mandatory = true, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = ParameterHelpMessages.ReportId)]
+        [ValidateNotNullOrEmpty]
+        public string ReportId { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.InitiativeNameScope, Mandatory = false, ValueFromPipelineByPropertyName = true, Position = 3, HelpMessage = ParameterHelpMessages.DetailedSwitch)]
+        [Parameter(ParameterSetName = ParameterSetNames.InitiativeIdScope, Mandatory = false, ValueFromPipelineByPropertyName = true, Position = 3, HelpMessage = ParameterHelpMessages.DetailedSwitch)]
+        [Parameter(ParameterSetName = ParameterSetNames.VmNameScope, Mandatory = false, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = ParameterHelpMessages.DetailedSwitch)]
+        public SwitchParameter Detailed { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.InitiativeNameScope, Mandatory = false, ValueFromPipelineByPropertyName = true, Position = 4, HelpMessage = ParameterHelpMessages.LatestSwitch)]
+        [Parameter(ParameterSetName = ParameterSetNames.InitiativeIdScope, Mandatory = false, ValueFromPipelineByPropertyName = true, Position = 4, HelpMessage = ParameterHelpMessages.DetailedSwitch)]
+        [Parameter(ParameterSetName = ParameterSetNames.VmNameScope, Mandatory = false, ValueFromPipelineByPropertyName = true, Position = 3,  HelpMessage = ParameterHelpMessages.DetailedSwitch)]
+        public SwitchParameter Latest { get; set; }
```

Comment:
same here

Reply:
- The below are the parameter sets. the last one does not have any switch. That means Optional parameters need parameter set names ?
Get-AzureRmVMGuestPolicyReport [-DefaultProfile <IAzureContextContainer>] [-Detailed] [-Latest] -ResourceGroupName <String> -VMName <String> [<CommonParameters>]

Get-AzureRmVMGuestPolicyReport [-DefaultProfile <IAzureContextContainer>] [-Detailed] -InitiativeId <String> [-Latest] -ResourceGroupName <String> -VMName <String> [<CommonParameters>]

Get-AzureRmVMGuestPolicyReport [-DefaultProfile <IAzureContextContainer>] [-Detailed] -InitiativeName <String> [-Latest] -ResourceGroupName <String> -VMName <String> [<CommonParameters>]

Get-AzureRmVMGuestPolicyReport [-DefaultProfile <IAzureContextContainer>] -ReportId <String> [<CommonParameters>]